ext/standard/array.c

```diff
@@ -4707,20 +4707,28 @@ PHP_FUNCTION(array_reduce)
 	}
 
 	fci.retval = &retval;
-	fci.param_count = 2;
+	fci.param_count = 3;
 	fci.no_separation = 0;
 
-	ZEND_HASH_FOREACH_VAL(htbl, operand) {
+	ZEND_HASH_FOREACH_BUCKET(htbl, operand) {
```

Comment:
That's a good point. No sense in dereferencing things from the bucket when there are macros to do it for you :).